example/index.tsx

```diff
@@ -4,13 +4,31 @@ import * as ReactDOM from 'react-dom';
 import ReactToPrint from '../src/index';
 import ComponentToPrint from './ComponentToPrint';
 
-class Example extends React.Component {
+interface State {
+    text: string;
+}
+
+interface Props {
+
+}
+
+class Example extends React.Component<Props, State> {
     componentRef: ComponentToPrint;
 
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            text: "000000000";
+        };
+    }
+
     handleAfterPrint = () => console.log('after print!');
     handleBeforePrint = () => console.log('before print!');
     renderContent = () => this.componentRef;
     renderTrigger = () => <button type="button">Print this out!</button>;
+    onBeforeGetContent = () => new Promise(resolve => {
+        this.setState({text: "text changed"}, () => resolve());
```

Comment:
Nit: `() => resolve()` can be changed to just `resolve`

Reply:
Please also handle the error case of the Promise rejecting (not that it would here, but the example should show users the full use-case)

Reply:
changed